### PR TITLE
[tests-only] Add sharing tests with permissions

### PR DESCRIPTION
**Pull request by SwikritiT**
This PR adds api tests for sharing with different permissions like read , upload , share ,change ,delete 
part of: https://github.com/owncloud/customgroups/issues/446

### CODE REVIEW

File: tests/acceptance/features/apiCustomGroups/shareOperationWithPermissions.feature [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    And the HTTP status code should be "200"
+    And as "Carol" folder "/shared" should exist
+    When user "Brian" shares file "/filetoshare.txt" with user "Carol" with permissions "share,read" using the sharing API
+    Then the OCS status code should be "100"
+    And the HTTP status code should be "200"
+    And as "Carol" file "/filetoshare.txt" should exist
+    And the content of file "/filetoshare.txt" for user "Carol" should be "thisIsASharedFile"
+
+
+  Scenario: User is not allowed to reshare folder/file with more permissions
```
Comment (Talank) [inline tests/acceptance/features/apiCustomGroups/shareOperationWithPermissions.feature:191]: does the `more permission` mean `share,update,read` and the `less permission` mean `share,read`?

Comment (SwikritiT) [reply tests/acceptance/features/apiCustomGroups/shareOperationWithPermissions.feature:191]: No, `more permission` can be any combination of permissions, it just has to be more compared to the `less permission` . Maybe I should test with all the possible combinations?

File: tests/acceptance/features/apiCustomGroups/shareOperationWithPermissions.feature [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    And user "Alice" has shared file "/filetoshare.txt" with group "customgroup_sharing-group" with permissions <received_permissions>
+    When user "Brian" shares file "/filetoshare.txt" with user "Carol" with permissions <reshare_permissions> using the sharing API
+    Then the OCS status code should be "100"
+    And the HTTP status code should be "200"
+    And the fields of the last response to user "Brian" sharing with user "Carol" should include
+      | share_with  | %username%            |
+      | file_target | /filetoshare.txt       |
+      | path        | /filetoshare.txt       |
+      | permissions | <granted_permissions> |
+      | uid_owner   | %username%            |
```
Comment (saw-jan) [inline tests/acceptance/features/apiCustomGroups/shareOperationWithPermissions.feature:239]: minor formatting!

File: tests/acceptance/features/apiCustomGroups/shareOperationWithPermissions.feature (hunk 1 above)
Comment (saw-jan) [reply tests/acceptance/features/apiCustomGroups/shareOperationWithPermissions.feature]: @SwikritiT 
the scenarios can be named as 
`Scenario: Custom group member is not allowed to reshare folder/file with permissions other than shared to them` Or something more fluent

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm 👍🏼

**Review by phil-davis — APPROVED** [review]

**Pull request merged**